pkg/urbit/noun/nock.c

```diff
@@ -1742,13 +1752,46 @@ _n_hilt_hind(u3_noun tok, u3_noun pro)
 static c3_o
 _n_hint_fore(u3_cell hin, u3_noun bus, u3_noun* clu)
 {
-  if ( c3__bout == u3h(hin) ) {
-    u3_atom now = u3i_chub(u3t_trace_time());
-    *clu = u3i_trel(u3h(hin), *clu, now);
-  }
-  else {
-    u3z(*clu);
-    *clu = u3_nul;
+  u3_noun tag, fol;
+  u3x_cell(hin, &tag, &fol);
+
+  switch ( tag ) {
+    case c3__bout: {
+      u3_atom now = u3i_chub(u3t_trace_time());
+      *clu = u3nt(u3k(tag), *clu, now);
+    } break;
+
+    case c3__nara: {
+      // unpack q_tok to get the priority integer and the tank
+      // pri is the priority, cap is the tank we will work with
+      u3_noun pri, cap;
+      c3_assert(c3y == u3r_cell(*clu, &pri, &cap));
+
+      // join the timing report with the original tank from cap like so:
+      // "cap: report"
+      // prepend the priority to form a cell of the same shape clu
+      // send this to ut3_slog so that it can be logged out
+      c3_l pri_l = c3y == u3a_is_cat(pri) ? pri : 0;
+      u3t_slog_cap(pri_l, u3i_string("trace of"), u3k(cap));
+      u3t_slog_nara(pri_l);
+      u3z(*clu);
+      *clu = u3_nul;
+    } break;
+
+    case c3__hela: {
+      u3_noun pri, cap;
+      c3_assert(c3y == u3r_cell(*clu, &pri, &cap));
+      c3_l pri_l = c3y == u3a_is_cat(pri) ? pri : 0;
+      u3t_slog_cap(pri_l, u3i_string("trace of"), u3k(cap));
+      u3t_slog_hela(pri_l);
+      u3z(*clu);
+      *clu = u3_nul;
```

Comment:
The assertion isn't appropriate here, since `*clu` contains arbitrary, user/dev controlled values. So it should be replaced with `if ( c3y == u3r_cell ...`. And `cap` is the wrong name for the tail -- `u3i_string("trace of")` is the caption). (both of these are my mistakes ...) 

The same goes for `%nara` above, *mutatis mutandis*.